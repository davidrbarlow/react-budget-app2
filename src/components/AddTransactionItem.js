import React from 'react';
import {connect} from 'react-redux';
import { Route , withRouter} from 'react-router-dom';
import TransactionItemForm from './TransactionItemForm';
import {startAddTransaction}  from '../actions/transactions';
import {toggleAddTransaction}  from '../actions/pageEdits';

export class AddTransactionItem extends React.Component {
 
  constructor(props) {
    super(props);
  
  }

    onSubmit = async (transaction) => {
      console.log('addtransitem', this.props);
        await this.props.startAddTransaction(transaction);
        // window.location.reload();
        await this.props.handleState();
        console.log('props in on submit ',this.props);
        this.props.toggleAddTransaction();
    };

    render() {
        return(
        <div>
            <TransactionItemForm 
            onSubmit={this.onSubmit}
            />
        </div>
    );
    }

}


const mapDispatchToProps = (dispatch) => ({
  startAddTransaction: (transaction) => dispatch(startAddTransaction(transaction)),
  toggleAddTransaction: () => dispatch(toggleAddTransaction())
})

export default connect(undefined, mapDispatchToProps)(withRouter(AddTransactionItem));