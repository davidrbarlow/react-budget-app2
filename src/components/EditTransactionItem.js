import React from 'react';
import {connect} from 'react-redux';
import TransactionItemForm from './TransactionItemForm';

const EditTransactionItem = (props) => {

  return (
    <div>
      <TransactionItemForms
        transaction = {props.transaction}
        onSubmit={(transaction)=>{
        props.dispatch(startEditTransaction(props.transaction.id, transaction));
        console.log(props.match.params.id);
      }}
      />


    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
      expense: state.transctions.find((transaction)=>transaction.id === props.match.params.id)
  };
};
export default connect(mapStateToProps)(EditExpensePage);