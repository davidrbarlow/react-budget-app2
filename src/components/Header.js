import React from 'react';
import {connect} from 'react-redux';
import {logout} from '../actions/auth.js';


class Header extends React.Component {
    constructor(props) {
        super(props);

        this.handleLogout = this.handleLogout.bind(this);

    }

    handleLogout = () => {
        const token = this.props.authToken;
        console.log('authToken is ',this.props.authToken);
        this.props.logout(token);
    }

    render() {
        return (
            <div className="header">
            Budget App
            <button className="header-button" onClick={this.handleLogout}>Logout</button>
        </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        logout: (token) => dispatch(logout(token))
    };
} ;

const mapStateToProps = (state) => {
    return {
        authToken: state.auth.authToken,
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Header);