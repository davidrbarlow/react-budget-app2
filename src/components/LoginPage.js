import React from 'react';
import {connect} from 'react-redux';
import {login} from '../actions/auth';
import {Link} from 'react-router-dom';


class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {};

        this.handleSubmit = this.handleSubmit.bind(this);
        this.autoComplete = 'yes';
        this.errorMessage = 'Username or Password is incorrect!'

    };

    onEmailChange = (e) =>{
        const email = e.target.value;
        this.setState(()=>({email}));
    };

    onPasswordChange = (e) =>{
        const password = e.target.value;
        this.setState(()=>({password}));
    };

    handleSubmit = (e) =>{
        e.preventDefault();
        let { email, password } = this.state;
        this.props.login(email, password);
        this.setState({
        email: '',
        password: ''
        });
        document.getElementById("login-form").reset();
  }


render(){
    //let {email, password} = this.state;
    let {isLoginPending, isLoginSuccess, loginError} = this.props;
    return(
        <div className="login-container">
            <div className="login-content-box">
                <div className="login-header">
                    <h1>Budget</h1>
                    <p>It's time to get your expenses under control.</p>
                </div>
                <div className="login">  
                    <form 
                    onSubmit={this.handleSubmit} id="login-form" 
                    autoComplete={this.autoComplete}
                    className="login-form">
                        
                        <div className="">
                        <input
                            type="text"
                            name="email"
                            placeholder="Email"
                            autoFocus={true}
                            className="input-text input-text__login"
                            //value={this.state.emal}
                            onChange={this.onEmailChange}
                        />  
                        </div>
                        <div>
                        <input
                            name="password"
                            type="text"
                            placeholder="Password"
                            className="input-text input-text__login"
                            //value={this.state.password}
                            onChange={this.onPasswordChange}
                        />
                        
                        </div>
                        <div className="login-page-sign-in-up">
                        <button className="button button__submit">
                        Login
                        </button>
                        <div>
                            { isLoginPending && <div>Please wait...</div> }
                            { isLoginSuccess && <div>Success.</div> }
                            { loginError && <div>{this.errorMessage}</div> }
                        </div>
                        <Link to="/signup">Sign up</Link>
                        </div>  
                    </form>
                </div>
            </div>
        </div>
        
        );
    };
};

const mapStateToProps = (state)=>{
    return {
    isLoginPending: state.isLoginPending,
    isLoginSuccess: state.isLoginSuccess,
    loginError: state.loginError,
    authToken: state.authToken
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        login: (email,password) => dispatch(login(email,password))
    };
};
// const mapDispatchToProps = (dispatch) => ({
//     startLogin2: () => dispatch(startLogin2())
// });



//export default connect(undefined, mapDispatchToProps)(LoginPage);

// cons mapDispatchToProps

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);

//export default LoginPage;

