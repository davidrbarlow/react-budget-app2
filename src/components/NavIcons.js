import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {setActivePage} from '../actions/pageEdits';

const NavIcons = (props) => (
    <div className="nav-icons">
      
    <Link to='dashboard'>
      <img id={'transactionsButton'} className="nav-icon"  src={'/images/transactions.svg'} 
      alt={'Transactions'} onClick={()=>{props.setActivePage('dashboard')}}></img>
    </Link>
      
      
    <Link to='/projection'> 
      <img id={'projectionButton'} className="nav-icon"  src={'images/projection.svg'} 
      alt={'Projections'} onClick={()=>{props.setActivePage('projection')}}
      ></img>
    </Link>
      
    </div>
);

const mapDispatchToProps = (dispatch) => ({
  setActivePage: (page) => dispatch(setActivePage(page))
});


export default connect(undefined, mapDispatchToProps)(NavIcons);

