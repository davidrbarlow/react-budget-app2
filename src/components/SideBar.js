import React from 'react';

const SideBar = () => (
    <div>
      <img id={'transactionsButton'} src={'/images/transactions.svg'} 
      alt={'Transactions'} width={'64'} height={'64'}></img>
      <img id={'projectionButton'} src={'images/projection.svg'} 
      alt={'Projections'}
      width ={'64'} height={'64'}></img>
    </div>
);

export default SideBar;
