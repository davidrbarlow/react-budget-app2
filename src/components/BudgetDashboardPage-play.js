import React from 'react';
import {connect} from 'react-redux';
import NavIcons from './NavIcons';
import Header from './Header';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';
import TransactionListControls from './TransactionListControls';

import { logout } from '../actions/auth';


class BudgetDashboardPage extends React.Component {

    constructor(props) {
        super(props);
        this.onUnload = this.onUnload.bind(this); // if you need to bind callback to this
    }

    componentWillUnmount() {
        // Make sure to remove the DOM listener when the component is unmounted.
        this.nv.removeEventListener("nv-enter", this.handleNvEnter);
      }

      componentDidMount() {
        // When the component is mounted, add your DOM listener to the "nv" elem.
        // (The "nv" elem is assigned in the render function.)
        this.nv.addEventListener("nv-enter", this.handleNvEnter);
      }

      handleNvEnter = (event) => {
        console.log("Nv Enter:", event);
      }


    onUnload = (event) => {
        console.log("hellooww");
        event.returnValue = "Hellooww";
        this.props.logout(this.props.token);
    }

    render() {
        return (
            <div className="layout">
                <div className="layout--top-left-corner"></div>
                <div className="layout--header">
                    <Header />
                </div>
                <div className="layout--nav-icon">
                    <NavIcons />
                </div>
                <div className="layout--filters">
                    <TransactionListFilters />
                </div>
            
                <div className="layout--controls">
                    <TransactionListControls />
                </div>
                <div className="layout--transactions">
                    <TransactionList />
                </div>       
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return{
        logout: (token) => dispatch(logout(token))
    };
} ;

const mapStateToProps = (state) => {
    return {
        authToken: state.auth.authToken,
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(BudgetDashboardPage);