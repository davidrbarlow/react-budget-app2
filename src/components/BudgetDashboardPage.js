import React from 'react';
import NavIcons from './NavIcons';
import Header from './Header';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';
import TransactionListControls from './TransactionListControls';


const BudgetDashboardPage = () => (
    <div className="layout">
        <div className="layout--top-left-corner"></div>
        <div className="layout--header">
            <Header />
        </div>
        <div className="layout--nav-icon">
            <NavIcons />
        </div>
        <div className="layout--filters">
            <TransactionListFilters />
        </div>
       
        <div className="layout--controls">
             <TransactionListControls />
        </div>
        <div className="layout--transactions">
            <TransactionList />
        </div>
        
        
        
    </div>
);

export default BudgetDashboardPage;