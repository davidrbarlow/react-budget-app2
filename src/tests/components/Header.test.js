import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {Header} from '../../components/Header';

test('shoud render Header correctly',()=> {
  
  const wrapper = shallow(<Header logout={()=>{}}/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
 
});