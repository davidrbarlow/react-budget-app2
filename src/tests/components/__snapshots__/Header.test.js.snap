// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render Header correctly 1`] = `
<div
  className="header"
>
  Budget App
  <button
    className="header-button"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
